concepts/functions/about.md

```diff
@@ -1,40 +1,167 @@
+import "fmt"
+
+// No parameters, no return value
+func PrintHello() {
+    fmt.Println("Hello")
+}
+// Called like this:
+PrintHello()
+
+// One parameter, one return value
+func Hello(name string) string {
+  return "Hello " + name
+}
+// Called like this:
+greeting := Hello("Dave")
+
+// Multiple parameters, multiple return values
+func SumAndMultiply(a, b int) (int, int) {
+    return a+b, a*b
+}
+// Called like this:
+aplusb, atimesb := SumAndMultiply(a, b)
+```
+
+## Named Return Values and Naked Return
+
+As well as parameters, return values can optionally be named.
+If named return values are used, a `return` statement without arguments will return those values, this is known as a 'naked' return.
+
+```go
+func SumAndMultiplyThenMinus(a, b, c int) (sum, mult int) {
+    sum, mult = a+b, a*b
+    sum -= c
+    mult -= c
+    return
+}
+```
+
+## Pointers
+
+Functions in go pass their arguments by value.
+This means that the arguments are copied, and any changes to those arguments within the function will not be seen outside the function.
+
+```go
+val := 2
+func MultiplyByTwo(v int) int {
+    v = v * 2
+    return v
+}
+newval := MultiplyByTwo(val)
+// newval is 4, val is 2
+```
+
+To affect the values a function is called with, it is possible to pass pointer arguments, using the `*` syntax in the function, and the `&` syntax to generate a pointer.
+
+```go
+func swap(x, y *int) {
+    var temp int
+    temp = *x
+    *x = *y
+    *y = temp
+}
+
+var a int = 100
+var b int = 200
+swap(&a, &b)
+// a = 200, b = 100
+```
+
+## Public and Private Functions
+
+To allow a function to be called by [code in other packages][public-vs-private], the name of the function must start with a capital letter.
 
 ```go
 package greeting
 
-// Hello is a public function
+// Hello is a public function (callable from other packages)
 func Hello(name string) string {
-    return hello(name)
+    return "Hello " + name
 }
 
-// hello is a private function
+// hello is a private function (not callable from other packages)
 func hello(name string) string {
     return "Hello " + name
 }
 ```
 
-Invoking a function from inside the same package is done by specifying the function name and passing arguments for each of the function's parameters.
+## Variadic Functions
+
+Go does not in general support optional parameters, but does support [variadic functions][variadic-functions] which can capture zero or more arguments at the end of a function call.
+
+This is expressed by prefixing the type of the last argument with `...`, like this:
+
+```go
+func sum(nums ...int) int {
+    total := 0
+    for _, num := range nums {
+        total += num
+    }
+    return total
+}
+```
 
-Invoking a function from another package is done by specifying the package name as well as the function name.
+A similar syntax can be used to unpack a slice into a function call:
 
 ```go
-phrase := greeting.Hello("John")
+numbers := []int{1, 2, 4}
+result := sum(...numbers)
+// result => 7
```

Comment:
Variadic functions are out of scope for this exercise (see the [issue description](https://github.com/exercism/go/issues/1642#issue-999186434)). Could you remove this section?